controllers/storage/provision.go

```diff
@@ -0,0 +1,366 @@
+package storage
+
+import (
+	goharborv1 "github.com/goharbor/harbor-cluster-operator/api/v1"
+	"github.com/goharbor/harbor-cluster-operator/lcm"
+	minio "github.com/minio/minio-operator/pkg/apis/operator.min.io/v1"
+	"strconv"
+
+	//minio "github.com/minio/minio-operator/pkg/apis/miniocontroller/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	Kind                        = "MinIOInstance"
+	ApiVersion                  = "miniooperator.min.io/v1beta1"
+	DefaultExternalSecretSuffix = "harbor-cluster-storage"
+	S3Secret                    = "s3Secret"
+	AzureSecret                 = "azureSecret"
+	GcsSecret                   = "gcsSecret"
+	SwiftSecret                 = "swiftSecret"
+	//DefaultCredsSecret          = "minio-creds-secret"
+	//DefaultMcsSecret            = "minio-mcs-secret"
+	//CredsAccesskey              = "bWluaW8="
+	//CredsSecretkey              = "bWluaW8xMjM="
+	//DefaultZone                 = "zone-harbor"
+)
+
+func (m *MinIOReconciler) ProvisionS3() (*lcm.Properties, error) {
+	s3Secret := m.generateS3Secret()
+	err := m.KubeClient.Create(m.Ctx, s3Secret)
+	p := &lcm.Property{
+		Name:  S3Secret,
+		Value: s3Secret.Name,
+	}
+	properties := &lcm.Properties{p}
+	return properties, err
+}
+
+func (m *MinIOReconciler) generateS3Secret() *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        m.HarborCluster.Name + "-" + DefaultExternalSecretSuffix,
+			Namespace:   m.HarborCluster.Namespace,
+			Labels:      m.getLabels(),
+			Annotations: m.generateAnnotations(),
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"region":         []byte(m.HarborCluster.Spec.Storage.S3.Region),
+			"bucket":         []byte(m.HarborCluster.Spec.Storage.S3.Bucket),
+			"accesskey":      []byte(m.HarborCluster.Spec.Storage.S3.Accesskey),
+			"secretkey":      []byte(m.HarborCluster.Spec.Storage.S3.Secretkey),
+			"regionendpoint": []byte(m.HarborCluster.Spec.Storage.S3.Regionendpoint),
+			"encrypt":        []byte(strconv.FormatBool(m.HarborCluster.Spec.Storage.S3.Encrypt)),
+			"keyid":          []byte(m.HarborCluster.Spec.Storage.S3.Keyid),
+			"secure":         []byte(strconv.FormatBool(m.HarborCluster.Spec.Storage.S3.Secure)),
+			"chunksize":      []byte(m.HarborCluster.Spec.Storage.S3.Chunksize),
+			"rootdirectory":  []byte(m.HarborCluster.Spec.Storage.S3.Rootdirectory),
+			"storageclass":   []byte(m.HarborCluster.Spec.Storage.S3.Storageclass),
+			"v4auth":         []byte(strconv.FormatBool(m.HarborCluster.Spec.Storage.S3.V4auth))},
```

Comment:
`Accesskey`，`Secretkey`, `Regionendpoint`... These words is not `CamelCase` naming.

Reply:
ref [here](https://github.com/goharbor/harbor-operator/blob/c8e922bded7080598ee5d10861271b792a22086d/docs/installation/registry-storage-configuration.md#create-the-secret)